### PR TITLE
Optimize deploying experience and adapt to a newer version of K8s(1.22+)

### PR DESCRIPTION
**Pull request by iamyeka**
Fix that horizon-core could not launch successfully if gitlab has been restarted && Bump Tekton trigger verson to 0.17.0 to fix the issue that EventListener cannot be deleted in helm uninstallation && Fix that some resources cannot be applied to v1.22+ K8s because some v1beta resource versions are unavailable in v1.22+

### CODE REVIEW

File: charts/horizon/charts/tektonci-resources/templates/horizon/trigger-el-new.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{{- if .Values.useNewTrigger }}
+apiVersion: triggers.tekton.dev/v1beta1
+kind: EventListener
+metadata:
```
Comment (bysph) [inline charts/horizon/charts/tektonci-resources/templates/horizon/trigger-el-new.yaml:4]: maybe use more clear file name, such as trigger-el-v1beta1

File: charts/gitlab/templates/configmap.yaml [hunk 2]
```diff
@@ -13,12 +13,24 @@ data:
...
+    # clear token to solve 500 error, refer: https://gitlab.com/gitlab-org/gitlab-foss/-/issues/59623,
+    # https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/26829/diffs
+    Project.update(runners_token: nil, runners_token_encrypted: nil)
+    Namespace.update(runners_token: nil, runners_token_encrypted: nil)
+
+    # create personal access token
+    token = User.find_by_username('root').personal_access_tokens.find_by_name('Automation token')
+    token.delete if token
+    token = User.find_by_username('root').personal_access_tokens.create(scopes: [:api], name: 'Automation token')
+    token.set_token('horizon-access-token')
```
Comment (bysph) [inline charts/gitlab/templates/configmap.yaml:35]: The plaintext token here is unsecure, it is better to generate token randomly, can we use other ways to solve it in the future?

File: charts/tektoncd/crds/extensions-crd.yaml [hunk 3]
```diff
@@ -1,27 +1,12 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
```
Comment (bysph) [inline charts/tektoncd/crds/extensions-crd.yaml:1]: does this apiVersion compatible with k8s 1.19?

File: charts/tektoncd/Chart.yaml [hunk 4]
```diff
@@ -3,8 +3,13 @@ name: tektoncd
...
-version: 2.0.1
+version: 2.1.0
 
-appVersion: 0.18.1
+appVersion: 0.28.0
 
 description: Helm chart for Tekton pipelines，Tekton triggers and Tekton dashboard.
+
+dependencies:
+  - name: common
```
Comment (kilosonc) [inline charts/tektoncd/Chart.yaml:13]: why this is nedded

File: charts/gitlab/Chart.yaml [hunk 5]
```diff
@@ -11,3 +11,8 @@ keywords:
 - code review
 - wiki
 home: https://about.gitlab.com
+
+dependencies:
+  - name: common
```
Comment (kilosonc) [inline charts/gitlab/Chart.yaml:16]: why this is needed?

Comment (iamyeka) [reply charts/gitlab/Chart.yaml:16]: Let ingress use the right version in different versions of k8s

**Review by kilosonc — APPROVED** [review]

**Review by bysph — APPROVED** [review]

**Pull request merged**